draft-ietf-quic-transport.md

```diff
@@ -1607,22 +1606,30 @@ NOT subsequently advertise a smaller offset.  A sender may receive WINDOW_UPDATE
 frames out of order; a sender MUST therefore ignore any reductions in flow
 control credit.
 
+A receiver MUST close the connection with a
+QUIC_FLOW_CONTROL_RECEIVED_TOO_MUCH_DATA error ({{error-handling}}) if the
+peer violates the advertised stream or connection flow control windows.
+
 A sender MUST send BLOCKED frames to indicate it has data to write but is
 blocked by lack of connection or stream flow control credit.  BLOCKED frames are
 expected to be sent infrequently in common cases, but they are considered useful
 for debugging and monitoring purposes.
 
 A receiver advertises credit for a stream by sending a WINDOW_UPDATE frame with
-the StreamID set appropriately.  A receiver may simply use the current received
-offset to determine the flow control offset to be advertised.
+the StreamID set appropriately. A receiver may use the current offset of data
+consumed to determine the flow control offset to be advertised.
+A receiver MAY send copies of a WINDOW_UPDATE frame in multiple packets in order
+to make sure that the sender receives it even if one of the packets is lost. A
+sender MUST ignore duplicate WINDOW_UPDATE frames.
```

Comment:
I like the sentence that @martinthomson suggested: "A WINDOW_UPDATE that doesn't move the window forward can be ignored".

@janaiyengar This PR doesn't change the part of the draft that says that WINDOW_UPDATEs MUST be retransmitted. In #114 we're having a discussion about relaxing this requirement in cases where the receiver can be sure that the sender already received a WINDOW_UPDATE with a greater offset. I really like that idea, but I think we should keep it out of this PR.